pkg/logql/syntax/ast_test.go

```diff
@@ -358,6 +358,34 @@ func Test_FilterMatcher(t *testing.T) {
 			},
 			[]linecheck{{"duration=5m total_bytes=5kB", true}, {"duration=1s total_bytes=256B", false}, {"duration=0s", false}},
 		},
+		{
+			`{app="foo"} |= "foo" or "bar"`,
+			[]*labels.Matcher{
+				mustNewMatcher(labels.MatchEqual, "app", "foo"),
+			},
+			[]linecheck{{"foo", true}, {"bar", true}, {"none", false}},
+		},
+		{
+			`{app="foo"} != "foo" or "bar"`,
+			[]*labels.Matcher{
+				mustNewMatcher(labels.MatchEqual, "app", "foo"),
+			},
+			[]linecheck{{"foo", false}, {"bar", false}, {"none", true}},
+		},
+		{
+			`{app="foo"} |~ "foo" or "bar"`,
```

Comment:
Can you also please add a test to verify that the filter type is copied to the `or` branch? i.e. if it's a regex matcher on the left, it should be a regex matcher on the right.

Reply:
Do we want to support this case? Regular expressions already come with alternatives.

Reply:
This is a good call. Supporting regex is a side effect of the way this is implemented and this test ensures it's correct. This actually found a bug where it wasn't always

Reply:
I agree with Danny. `|~ "foo|bar" or "\d+baz"` Eg would match a string containing "foo", "bar" or "42baz". This makes sense to me.